receiver/podmanreceiver/metrics.go

```diff
@@ -0,0 +1,143 @@
+	sum(ms, "network.io.usage.tx_bytes", "By", []point{{intVal: stats.NetInput}}, ts)
+	sum(ms, "network.io.usage.rx_bytes", "By", []point{{intVal: stats.NetOutput}}, ts)
+}
+
+func appendIOMetrics(ms pdata.MetricSlice, stats *define.ContainerStats, ts pdata.Timestamp) {
+	sum(ms, "blockio.io_service_bytes_recursive.write", "By", []point{{intVal: stats.BlockOutput}}, ts)
+	sum(ms, "blockio.io_service_bytes_recursive.read", "By", []point{{intVal: stats.BlockInput}}, ts)
+}
+
+func appendCPUMetrics(ms pdata.MetricSlice, stats *define.ContainerStats, ts pdata.Timestamp) {
+	sum(ms, "cpu.usage.system", "ns", []point{{intVal: stats.CPUSystemNano}}, ts)
+	sum(ms, "cpu.usage.total", "ns", []point{{intVal: stats.CPUNano}}, ts)
+	gaugeF(ms, "cpu.percent", "1", []point{{doubleVal: stats.CPU}}, ts)
+
+	points := make([]point, len(stats.PerCPU))
+	for i, cpu := range stats.PerCPU {
+		points[i] = point{
+			intVal: cpu,
+			labels: map[string]string{
+				"core": fmt.Sprintf("cpu%d", i),
+			},
+		}
+	}
+	sum(ms, "cpu.usage.percpu", "ns", points, ts)
+}
+
+func initMetric(ms pdata.MetricSlice, name, unit string) pdata.Metric {
+	m := ms.AppendEmpty()
+	m.SetName(fmt.Sprintf("container.%s", name))
+	m.SetUnit(unit)
+	return m
+}
+
+func sum(ilm pdata.MetricSlice, metricName string, unit string, points []point, ts pdata.Timestamp) {
+	metric := initMetric(ilm, metricName, unit)
+
+	metric.SetDataType(pdata.MetricDataTypeSum)
+	sum := metric.Sum()
+	sum.SetIsMonotonic(true)
+	sum.SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
+
+	dataPoints := sum.DataPoints()
+
+	for _, pt := range points {
+		dataPoint := dataPoints.AppendEmpty()
+		dataPoint.SetTimestamp(ts)
+		dataPoint.SetIntVal(int64(pt.intVal))
+		setDataPointLabels(dataPoint, pt.labels)
+	}
+}
+
+func gauge(ms pdata.MetricSlice, metricName string, unit string, ts pdata.Timestamp) pdata.NumberDataPointSlice {
+	metric := initMetric(ms, metricName, unit)
+	metric.SetDataType(pdata.MetricDataTypeGauge)
+
+	gauge := metric.Gauge()
+	return gauge.DataPoints()
+}
+
+func gaugeI(ms pdata.MetricSlice, metricName string, unit string, points []point, ts pdata.Timestamp) {
+	dataPoints := gauge(ms, metricName, unit, ts)
+	for _, pt := range points {
+		dataPoint := dataPoints.AppendEmpty()
+		dataPoint.SetTimestamp(ts)
+		dataPoint.SetIntVal(int64(pt.intVal))
+		setDataPointLabels(dataPoint, pt.labels)
+	}
+}
+
+func gaugeF(ms pdata.MetricSlice, metricName string, unit string, points []point, ts pdata.Timestamp) {
+	dataPoints := gauge(ms, metricName, unit, ts)
+	for _, pt := range points {
+		dataPoint := dataPoints.AppendEmpty()
+		dataPoint.SetTimestamp(ts)
+		dataPoint.SetDoubleVal(pt.doubleVal)
+		setDataPointLabels(dataPoint, pt.labels)
+	}
+}
+
+func setDataPointLabels(dataPoint pdata.NumberDataPoint, labels map[string]string) {
+	for k, v := range labels {
+		dataPoint.LabelsMap().Insert(k, v)
```

Comment:
We are converting all components to use data point Attributes instead of Labels. Perhaps use Attributes right away? Labels are obsolete.

Reply:
I'll look into that. Thanks

Reply:
Updated